pkg/certificate/providers/config.go

```diff
@@ -29,227 +29,153 @@ const (
-
 	rootCert, err = tresor.NewCA(constants.CertificationAuthorityCommonName, constants.CertificationAuthorityRootValidityPeriod, rootCertCountry, rootCertLocality, rootCertOrganization)
-
 	if err != nil {
-		return nil, nil, errors.Errorf("Failed to create new Certificate Authority with cert issuer %s", c.providerKind)
-	}
-
-	if rootCert == nil {
-		return nil, nil, errors.Errorf("Invalid root certificate created by cert issuer %s", c.providerKind)
+		return nil, errors.New("Failed to create new Certificate Authority with cert issuer tresor")
 	}
 
 	if rootCert.GetPrivateKey() == nil {
-		return nil, nil, errors.Errorf("Root cert does not have a private key")
+		return nil, errors.New("Root cert does not have a private key")
 	}
 
-	rootCert, err = k8s.GetCertificateFromSecret(c.providerNamespace, c.caBundleSecretName, rootCert, c.kubeClient)
+	rootCert, err = k8s.GetCertificateFromSecret(mrc.Namespace, mrc.Spec.Provider.Tresor.SecretName, rootCert, c.kubeClient)
 	if err != nil {
-		return nil, nil, errors.Errorf("Failed to synchronize certificate on Secrets API : %v", err)
+		return nil, fmt.Errorf("Failed to synchronize certificate on Secrets API : %w", err)
 	}
 
 	if rootCert.GetPrivateKey() == nil {
-		return nil, nil, fmt.Errorf("Root cert does not have a private key: %w", certificate.ErrInvalidCertSecret)
+		return nil, fmt.Errorf("Root cert does not have a private key: %w", certificate.ErrInvalidCertSecret)
 	}
 
 	tresorClient, err := tresor.New(
 		rootCert,
 		rootCertOrganization,
-		c.cfg.GetCertKeyBitSize(),
+		c.KeyBitSize,
 	)
 	if err != nil {
-		return nil, nil, errors.Errorf("Failed to instantiate Tresor as a Certificate Manager")
-	}
-
-	tresorCertManager, err := certificate.NewManager(tresorClient, c.cfg.GetServiceCertValidityPeriod(), c.msgBroker)
-	if err != nil {
-		return nil, nil, fmt.Errorf("error instantiating osm certificate.Manager for Tresor cert-manager : %w", err)
+		return nil, fmt.Errorf("failed to instantiate Tresor as a Certificate Manager: %w", err)
 	}
-	return tresorCertManager, tresorCertManager, nil
+	return tresorClient, nil
 }
 
 // getHashiVaultOSMCertificateManager returns a certificate manager instance with Hashi Vault as the certificate provider
-func (c *Config) getHashiVaultOSMCertificateManager(options VaultOptions) (*certificate.Manager, debugger.CertificateManagerDebugger, error) {
-	if _, ok := map[string]interface{}{"http": nil, "https": nil}[options.VaultProtocol]; !ok {
-		return nil, nil, fmt.Errorf("value %s is not a valid Hashi Vault protocol", options.VaultProtocol)
+func (c *MRCProviderGenerator) getHashiVaultOSMCertificateManager(mrc *v1alpha2.MeshRootCertificate) (certificate.Issuer, error) {
+	provider := mrc.Spec.Provider.Vault
+	if _, ok := map[string]interface{}{"http": nil, "https": nil}[provider.Protocol]; !ok {
+		return nil, fmt.Errorf("value %s is not a valid Hashi Vault protocol", provider.Protocol)
 	}
```

Comment:
Why does this need to be validated here if it is already validated in `validate()` for Hashi Vault?

Reply:
good catch! removed